unlock-app/src/components/interface/checkout/Connect/index.tsx

```diff
@@ -1,23 +1,25 @@
 import React, { useCallback, useMemo } from 'react'
 import { useCheckoutCommunication } from '~/hooks/useCheckoutCommunication'
 import type { OAuthConfig } from '~/unlockTypes'
-import { ConfirmConnect } from './Confirm'
 import { connectMachine } from './connectMachine'
-import { UnlockAccountSignIn } from './UnlockAccountSignIn'
 import { CheckoutHead, TopNavigation } from '../Shell'
 import { useMachine } from '@xstate/react'
+import ConnectWalletComponent from '../../connect/ConnectWalletComponent'
+import { generateNonce } from 'siwe'
+import { useSIWE } from '~/hooks/useSIWE'
+import { useAuth } from '~/contexts/AuthenticationContext'
+import { PaywallConfigType } from '@unlock-protocol/core'
 
 interface Props {
   oauthConfig: OAuthConfig
-  injectedProvider: unknown
+  paywallConfig: PaywallConfigType
 }
 
-export function Connect({ injectedProvider, oauthConfig }: Props) {
+export function Connect({ paywallConfig, oauthConfig }: Props) {
   const communication = useCheckoutCommunication()
 
   // @ts-expect-error - The types returned by 'resolveState(...)' are incompatible between these types
   const [state, send, connectService] = useMachine(connectMachine)
```

Comment:
I think we should get rid of the connectMachine. It has 2 states only:
* Connecting (either with wallet or unlock account)
* Conforming that the user is ok to share data

No need to make it a full state machine TBH.